src/legacy/core_plugins/kibana/public/management/sections/index_patterns/create_index_pattern_wizard/components/step_time_field/step_time_field.js

```diff
@@ -109,10 +118,45 @@ export class StepTimeFieldComponent extends Component {
     }));
   }
 
-  createIndexPattern = () => {
+  createIndexPattern = async () => {
+    const { createIndexPattern } = this.props;
     const { selectedTimeField, indexPatternId } = this.state;
     this.setState({ isCreating: true });
-    this.props.createIndexPattern(selectedTimeField, indexPatternId);
+    try {
+      await createIndexPattern(selectedTimeField, indexPatternId);
+    } catch (error) {
+      if (!this.mounted) return;
+      this.setState({
+        error: error instanceof Error ? error.message : String(error)
+      });
+    } finally {
+      if (this.mounted) {
+        this.setState({ isCreating: false });
+      }
+    }
+  }
+
+  formatErrorMessage(message) {
+    // `createIndexPattern` throws "Conflict" when index pattern ID already exists.
```

Comment:
This feels like a pretty fragile way to check for a conflict error. Is this the only way? Is there a more structured way, such as checking an error code, or something? Otherwise, this is probably good enough, just feels hacky!

Reply:
Yes, it just returns `"Conflict"` 🤷‍♂️

![image](https://user-images.githubusercontent.com/9773803/57456123-0e64ce80-726d-11e9-84e1-38ffd4f07a3e.png)

Reply:
I dug into the code and think its a good hack. Getting a proper error message would require a much larger PR.